jvb/src/main/kotlin/org/jitsi/videobridge/EndpointConnectionStatusMonitor.kt

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.videobridge
+
+import org.jitsi.nlj.util.NEVER
+import org.jitsi.utils.logging2.Logger
+import org.jitsi.utils.logging2.cdebug
+import org.jitsi.utils.logging2.createChildLogger
+import org.jitsi.videobridge.message.EndpointConnectionStatusMessage
+import java.time.Clock
+import java.time.Duration
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicReference
+
+class EndpointConnectionStatusMonitor @JvmOverloads constructor(
+    private val conference: Conference,
+    private val executor: ScheduledExecutorService,
+    parentLogger: Logger,
+    private val clock: Clock = Clock.systemUTC()
+) {
+    private val logger = createChildLogger(parentLogger)
+
+    private val config = EndpointConnectionStatusConfig()
+
+    private val inactiveEndpointIds = mutableSetOf<String>()
+
+    private val taskHandle = AtomicReference<ScheduledFuture<*>>(null)
+
+    fun start() {
+        if (taskHandle.compareAndSet(
+                null,
+                executor.scheduleWithFixedDelay(::run, config.intervalMs, config.intervalMs, TimeUnit.MILLISECONDS))
+        ) {
+            logger.info("Starting connection status monitor")
+        } else {
+            logger.warn("Task already started, not starting again")
```

Comment:
is this check just in case or is it normal to call start more than once?

Reply:
Not normal, just a just-in-case check